pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -175,112 +176,82 @@ type ReconcileCnsNodeVMAttachment struct {
 // will remove the work from the queue.
 func (r *ReconcileCnsNodeVMAttachment) Reconcile(ctx context.Context,
 	request reconcile.Request) (reconcile.Result, error) {
-	ctx = logger.NewContextWithLogger(ctx)
-	log := logger.GetLogger(ctx)
-	// Fetch the CnsNodeVmAttachment instance
-	instance := &cnsnodevmattachmentv1alpha1.CnsNodeVmAttachment{}
-	err := r.client.Get(ctx, request.NamespacedName, instance)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Error("CnsNodeVmAttachment resource not found. Ignoring since object must be deleted.")
-			return reconcile.Result{}, nil
+	start := time.Now()
+	volumeType := prometheus.PrometheusBlockVolumeType
+	var volumeOpType string
+	reconcileCnsNodeVMAttachmentInternal := func() (
+		reconcile.Result, error) {
+		ctx = logger.NewContextWithLogger(ctx)
+		log := logger.GetLogger(ctx)
+		// Fetch the CnsNodeVmAttachment instance
+		instance := &cnsnodevmattachmentv1alpha1.CnsNodeVmAttachment{}
+		volumeOpType = prometheus.PrometheusAttachVolumeOpType
```

Comment:
Looks like we are using the same volumeOperationType for Vanilla, Supervisor and TKG cluster. Are these metrics viewed per VC or per kubernetes cluster. Let's say I have a VC with all 3 cluster types, I am trying to understand will the metrics collide if I want to view it per VC ? Can you confirm

Reply:
First, I don't think a VC has all 3 cluster type is a common configuration that customer will have. 
The metrics is exposed as per kubernetes cluster. We defined a CSI service with LoadBalancer type, and the CSI metrics is exposed from the service.

Reply:
@lipingxue to clarify, we do not prevent a customer from deploying all flavors of k8s clusters we support in vCenter. 

@chethanv28 The Prometheus server(or metrics scraper)can be configured to add any extra labels that is unique to the k8s cluster deployment(like clutser flavor, cluster name, etc) if necessary. It wont make sense to add such details to all the CSI metrics by default, I think it is redundant.